libass/ass.c

```diff
@@ -1520,19 +1520,43 @@ ASS_Track *ass_new_track(ASS_Library *library)
     return NULL;
 }
 
-int ass_track_set_feature(ASS_Track *track, ASS_Feature feature, int enable)
-{
-    switch (feature) {
-    case ASS_FEATURE_INCOMPATIBLE_EXTENSIONS:
-        //-fallthrough
+static const uint64_t supported_features =
 #ifdef USE_FRIBIDI_EX_API
-    case ASS_FEATURE_BIDI_BRACKETS:
-        track->parser_priv->bidi_brackets = !!enable;
+    1LL << ASS_FEATURE_BIDI_BRACKETS |
 #endif
-        return 0;
-    default:
+    0;
+
+int ass_track_set_feature(ASS_Track *track, ASS_Feature feature, int enable)
+{
+    uint64_t requested;
```

Comment:
We're using C99, so it's better to combine variable definition with its first use. Also type inconsistency.